api/app/backend_proxy.rb

```diff
@@ -60,22 +60,32 @@ def rewrite_response(triplet)
 
   private
 
-  def auth_from_cookies(env)
+  def credentials_from_cookies(env)
     request_cookies = Rack::Utils.parse_cookies(env)
-    request_cookies['auth_token'].chomp
+    request_cookies[FlightFileManager.app.config.cloudcmd_cookie_name].chomp
   end
 
   def current_user(env)
-    credentials = auth_from_cookies(env).split(' ').last
+    credentials = credentials_from_cookies(env)
     username, _ = Base64.decode64(credentials).split(':', 2)
     username
   end
 
   def assert_good_origin(env)
     request = Rack::Request.new(env)
+
+    # Work around Firefox sometimes not including the referer/origin header.
+    # But only for GET requests.
+    if request.referer.nil?
```

Comment:
It is possible `rack`/`sinatra` is doing something funky in the background, so this may not apply.

However does `request.referer` check the `origin` header as well? IIRC there are a couple of browser edge cases where `origin` is set but not `referrer`.

Reply:
The rack request object doesn't do anything with the `origin` header FSR.  I suspect that it has a good reason not to as many of its other methods are setup to handle alternate headers, proxying and proxying HTTPS to HTTP.